src/App.tsx

```diff
@@ -2,15 +1,54 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Dropdown } from './components/Dropdown';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [focused, setFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useMemo(() => debounce(setQuery, 300), []);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(query.toLowerCase()),
+    );
+  }, [query]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedPerson(null);
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
The `applyQuery` function is called immediately after `setQuery`, which means the query is updated twice: once immediately and once after the debounce delay. This defeats the purpose of debouncing. You should decide whether to update the query immediately or use debouncing to delay the update, but not both.